### PR TITLE
Issue/2965 reader read sites endpoints

### LINKED ISSUE #2965: Reader: use read/sites endpoints
When retrieving Reader posts, use the new v1.1 `read/sites` endpoints that avoid hitting the Jetpack proxy. Specifically:

* `read/sites/{siteId}`
* `read/sites/{siteId}/posts`
* `read/sites/{siteId}/posts/{postId}`

### PR DESCRIPTION
**Pull request by nbradbury**
Resolves #2965 - Updates the reader to use the `read/sites/` rather than `sites/` endpoints when retrieving posts. These endpoints provide performance benefits when accessing Jetpack blogs/posts.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/datasets/ReaderDatabase.java [hunk 1]
```diff
@@ -53,6 +53,7 @@
      *   99 - added feed_url to tbl_blog_info
      *  100 - changed primary key on tbl_blog_info
      *  101 - dropped is_reblogged from ReaderPostTable
+     *  102 - changed primary key of tbl_blog_info from blog_id+feed_id to just blog_id
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/datasets/ReaderDatabase.java]: What is the reason behind this change?

**Comment by daniloercoli** [comment]
Is there any reason why the `read/sites/` endpoint is not being documented on developer.wordpress.com?

File: WordPress/src/main/java/org/wordpress/android/datasets/ReaderDatabase.java (hunk 1 above)
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/datasets/ReaderDatabase.java]: > What is the reason behind this change?

Sorry, I should've explained that above. The old calls would return a `blog_id` of zero for all feeds, and since these had a non-zero `feed_id` it was necessary to make the primary key include both those columns. The new calls *do* return a `blog_id` for feeds, so the primary key only needs to be on `blog_id`.

**Comment by nbradbury** [comment]
> Is there any reason why the read/sites/ endpoint is not being documented on developer.wordpress.com?

These endpoints are really a temporary stop-gap until the main `sites/` endpoints do what we want. Once that occurs (which may take a while) we'll switch back to them and drop these reader-specific ones. So it made sense not to document them since we don't really want people to start using endpoints that we're only going to yank later on.

**Comment by daniloercoli** [comment]
:shipit:
